src/Illuminate/Validation/ValidationRuleParser.php

```diff
@@ -139,6 +132,10 @@ protected function prepareRule($rule, $attribute)
             )->rules[$attribute];
         }
 
+        if (str_contains((string) $rule, '|')) {
+            return explode('|', (string) $rule);
+        }
+
         return (string) $rule;
```

Comment:
You could avoid having to cast the `$rule` to string three times:
```suggestion

        $rule = (string) $rule;

        if (str_contains($rule, '|')) {
            return explode('|', $rule);
        }

        return $rule;
```